### PR TITLE
Contributing and Project Guidelines overhaul (again)

### PR DESCRIPTION
**Pull request by kmcculloch**
Since @Mr0grog had some proposed changes, I reverted the previous merge and cherry-picked @dcwalk 's work onto this new branch. Sorry for all the mess. I'll stop messing with this repo now.

@b5 @danielballan @lightandluck @titaniumbones 

Please see @Mr0grog 's responses to the original PR here:

https://github.com/edgi-govdata-archiving/overview/pull/89

### CODE REVIEW

**Comment by dcwalk** [comment]
No worries @kmcculloch, thanks for reverting. 
I've added in the changes from @Mr0grog to this branch.

**Comment by dcwalk** [comment]
@Mr0grog -- totally agree about adding in a copyright block!! Here is a GPL v3 template:

```
/**
 * Some open source application
 *
 *  Copyright 2012 by Tobias Eriksson <author@tobier.se>
 *  Copyright 2010, 2011 by Scruffy H. Hacker <scruffy@foo.bar>
 *
 *  Licensed under GNU General Public License 3.0 or later. 
 *  Some rights reserved. See COPYING, AUTHORS.
 *
 * @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
 */
```

**Comment by Mr0grog** [comment]
@dcwalk Oh, I was thinking about descriptive text to go in the README that links to the LICENSE file, not as the header for source files (which, obviously, I should *also* have been thinking of e1b6354bbbba3450 )

**Comment by Mr0grog** [comment]
e.g:

```md
## License & Copyright

This project is copyright (c) 2017 The Environmental Data and Governance Initiative.
All source code is licensed under the GPL v3 source code license, found in the
[`LICENSE`](https://github.com/edgi-govdata-archiving/<repo name here>/blob/master/LICENSE)
file.
```

**Comment by dcwalk** [comment]
Yeah, I couldn't find one of those-- but though we could squash the above into yours!

**Comment by Mr0grog** [comment]
Oh, heh, I misunderstood. What do you think of…

Somewhere in a project Readme:

```md
## License & Copyright

Copyright (C) <year> The Environmental Data and Governance Initiative
This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

See the [`LICENSE`](https://github.com/edgi-govdata-archiving/<repo name here>/blob/master/LICENSE) file for details.
```

And the suggested source code header (looks like recommended language changed from the 3.0+ one to 3.0 only):

```
/**
 * Copyright (C) <year(s)> The Environmental Data and Governance Initiative
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, version.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the `LICENSE` file for details.
 */
```

**Comment by Mr0grog** [comment]
Hmmm, actually, that one you posted is a little easier to read and less dense. So…?

```
/**
 *  Copyright <year(s)> by The Environmental Data and Governance Initiative
 *  Copyright <year> by <your name> <your@email.com>
 *
 *  Licensed under GNU General Public License 3.0. 
 *  See LICENSE file for details.
 *
 * @license GPL-3.0 <http://spdx.org/licenses/GPL-3.0>
 */
```

**Comment by dcwalk** [comment]
I actually think the fuller one is probably better? I've pushed some templates, let me know what you all think?

**Comment by dcwalk** [comment]
This will address our recommending practises for licenses: #76

**Comment by dcwalk** [comment]
Okay, included a bit of @kmcculloch work on the linting proposal, but honestly I almost want to bring it all here. I think super great and totally a project guideline!  (also-- all this makes me feel like we gotta get up to snuff on all our projects ;)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,27 +1,43 @@
 # Contributing Guidelines
 
-We love improvements to our tools! Our process for accepting changes has a few steps:
+We love improvements to our tools! There are a few key ways people can help us improve our projects:
```
Comment (Mr0grog) [inline CONTRIBUTING.md]: “People can help” → “you can help” might read nicer here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
+Our process for accepting changes operates by [Pull Request (PR)](https://help.github.com/articles/about-pull-requests/) and has a few steps:
+
+1.  If you haven't submitted anything before, and you aren't (yet!) a member of our organization, **fork and clone** the repo:
 
         $ git clone git@github.com:<your-username>/<repository-name>.git
 
-  Organization members should clone the upsteam repo, instead of working from a personal fork:
+    Organization members should clone the upsteam repo, instead of working from a personal fork:
 
       $ git clone git@github.com:edgi-govdata-archiving/<repository-name>.git
```
Comment (Mr0grog) [inline CONTRIBUTING.md]: Obviously not something you introduced, but I noticed there is one too few spaces at the start of this line, which is causing it not to get treated as a code block.

File: PROJECT.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Project Guidelines
+
+Each project repository **require**, at a minimum:
```
Comment (Mr0grog) [inline PROJECT.md]: “require” → “requires”

**Review by Mr0grog — COMMENTED** [review]
This is awesome. Spotted some minor typos, but otherwise looks wonderful to me 👍

**Comment by dcwalk** [comment]
Hey @danielballan or @titaniumbones could you give this a once over before merging in?

**Comment by titaniumbones** [comment]
Made some very minor language changes. I'm good to merge this now, but will let you take a quick look at my changes first @dcwalk .

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,27 +1,43 @@
...
+We have [project guidelines](./PROJECT.md) for all of the projects hosted in our [GitHub Organization](https://github.com/edgi-govdata-archiving), which new repositories should follow during their creation.
+
+Our process for accepting changes operates by [Pull Request (PR)](https://help.github.com/articles/about-pull-requests/) and has a few steps:
+
+1.  If you haven't submitted anything before, and you aren't (yet!) a member of our organization, **fork and clone** the repo:
 
         $ git clone git@github.com:<your-username>/<repository-name>.git
 
-  Organization members should clone the upsteam repo, instead of working from a personal fork:
+    Organization members should clone the upsteam repo, instead of working from a personal fork:
```
Comment (danielballan) [inline CONTRIBUTING.md:25]: Just out of curiosity, what is the rationale for org members pushing working branches to the upstream fork? To make long-lived branches easier to discover?

Comment (dcwalk) [reply CONTRIBUTING.md:25]: Definitely discovery to figure out who is working on stuff (I've worked on some slooooow moving projects) but also I it seems to have a nice rhythm for for collaborating (and feeling able to collaborate) on work by multiple people pushing commits.
^^ (when asked :))

**Comment by dcwalk** [comment]
Thanks all for your eyes. I'm merging in and we can obviously revisit as needed!

File: CONTRIBUTING.md (hunk 4 above)
Comment (Mr0grog) [reply CONTRIBUTING.md:25]: Would echo that last point about multiple people working on a branch. A total no-no in some groups, but *wonderful* in others when it works out well and people are open to it. Hard to do if everyone maintains their own fork.

Comment (danielballan) [reply CONTRIBUTING.md:25]: Interesting. As a project maintainer I have enjoyed GH's new feature allowing me to push to others' PR branches (if they allow it). This seems like a similar workflow, but no one "owns" the branch.

Comment (Mr0grog) [reply CONTRIBUTING.md:25]: Ah! I have not used that at all, though I do remember seeing a button for that recently.

Comment (dcwalk) [reply CONTRIBUTING.md:25]: yeah the push-as-maintainer is nice and I've used it recently here. But it feels a little less communal. 
I dunno, call me a collectivist or something-- I really dig co-working on branches with people :)

**Pull request merged**